### PR TITLE
Add CodeQL Security Scanning Workflow for AOC

### PR DESCRIPTION
**Pull request by JasonXZLiu**
**Description:** 

This PR adds basic CodeQL scanning into the CI and PR-Build workflows to perform security scanning on the AWS-OTel-Collector. CodeQL will create alerts based on it's analysis which will be found in the security tab: https://github.com/aws-observability/aws-otel-collector/security.

### CODE REVIEW

**Review by alolita — APPROVED** [review]
+1

**Review by amanbrar1999 — APPROVED** [review]

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -94,7 +93,32 @@ jobs:
...
+
+    steps:
+      - uses: actions/checkout@v2
 
+      - name: Initialize CodeQL
+        uses: github/codeql-action/init@v1
+        with:
+          languages: "go"
+        
+      - name: Download built artifacts
```
Comment (wyTrivail) [inline .github/workflows/CI.yml:110]: Why do we need to download the artifacts?

File: .github/workflows/PR-build.yml [hunk 2]
```diff
@@ -61,6 +61,31 @@ jobs:
...
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Initialize CodeQL
+        uses: github/codeql-action/init@v1
+        with:
+          languages: "go"
+        
+      - name: Download built artifacts
```
Comment (wyTrivail) [inline .github/workflows/PR-build.yml:77]: the same question, i thought it's just a source code scan?

File: .github/workflows/CI.yml (hunk 1 above)
Comment (JasonXZLiu) [reply .github/workflows/CI.yml:110]: Hmm, I think by default, some languages must be built and compiled for the code scanning (more specifically for languages like Java and C++). I wasn't sure if we needed it for Golang (but I haven't found any documentation on it). I'll try testing without it and seeing if the code scanning still runs

Comment (JasonXZLiu) [reply .github/workflows/CI.yml:110]: Seems like we actually don't need it. I'll remove this and push again.

**Review by wyTrivail — APPROVED** [review]

**Pull request merged**